classes/Api/ThankYous.php

```diff
@@ -0,0 +1,809 @@
+	 * @throws ThankYouForbidden - If the Security Context's User does not have permission.
+	 */
+	public function Delete(SecurityContext $security_context, int $id)
+	{
+		$thank_you = $this->GetThankYou($id);
+
+		if (!$this->CanDeleteThankYou($thank_you, $security_context))
+		{
+			throw new ThankYouForbidden("Failed to Update Thank You, User is not the Author and does not have administrative privileges");
+		}
+
+		$this->tag_api->RemoveAllTaggableTaggings($id, ThankYousRepository::AGGREGATION_ID);
+		$this->thank_yous_repository->Delete($id);
+
+		$this->audit->Store(Audit::AUDIT_SUCCESS, Plugin::APPLICATION_NAME, 'thank_you_delete', $id, $thank_you->GetDescription());
+	}
+
+	/**
+	 * @param ThankYou        $thank_you
+	 * @param SecurityContext $security_context
+	 * @return bool
+	 */
+	public function CanDeleteThankYou(ThankYou $thank_you, SecurityContext $security_context)
+	{
+		return $this->acl->CanDeleteThankYou($thank_you, $security_context);
+	}
+
+	/**
+	 * @param ThankYou        $thank_you
+	 * @param SecurityContext $security_context
+	 * @return bool
+	 */
+	public function CanEditThankYou(ThankYou $thank_you, SecurityContext $security_context)
+	{
+		return $this->acl->CanEditThankYou($thank_you, $security_context);
+	}
+
+	/**
+	 * @param SecurityContext $security_context
+	 * @param ThankYou        $thank_you
+	 * @return bool
+	 */
+	public function CanSeeThankYouAuthor(SecurityContext $security_context, ThankYou $thank_you): bool
+	{
+		return $this->acl->CanSeeThankYouAuthor($security_context, $thank_you);
+	}
+
+	/**
+	 * @param SecurityContext $security_context
+	 * @param Thankable       $thankable
+	 * @return bool
+	 */
+	public function CanSeeThankableName(SecurityContext $security_context, Thankable $thankable): bool
+	{
+		return $this->acl->CanSeeThankableName($security_context, $thankable);
+	}
+
+	/**
+	 * Determines whether a Security Context can view a User's details.
+	 *
+	 * @param SecurityContext $context
+	 * @param User            $user
+	 * @return bool
+	 */
+	public function CanSeeUser(SecurityContext $context, User $user): bool
+	{
+		return $this->acl->CanSeeUser($context, $user);
+	}
+
+	/**
+	 * @param SecurityContext $security_context
+	 * @return bool
+	 */
+	public function IsAdmin(SecurityContext $security_context): bool
+	{
+		return $this->acl->IsAdmin($security_context);
+	}
+
+	/**
+	 * Determines whether an Extranet Area is visible. If the second parameter is provided, this will be relative to
+	 * that Extranet.
+	 *
+	 * @param int      $target_extranet_id
+	 * @param int|null $viewers_extranet_id
+	 * @return bool
+	 */
+	public function IsExtranetVisible(int $target_extranet_id, ?int $viewers_extranet_id = null): bool
+	{
+		return $this->acl->IsExtranetVisible($target_extranet_id, $viewers_extranet_id);
+	}
+
+	/**
+	 * @param ThankYou $thank_you
+	 */
+	public function PopulateThankYouUsersFromThankables(ThankYou $thank_you)
+	{
+		$thankables = $thank_you->GetThankable();
```

Comment:
Tiny thing: The method could be renamed to `GetThankables`. Makes it clearer that it'll return a collection.

Reply:
Fixed in a86f4e6a3fb39fd21c9c4c74cba83b4ece943bc7